string_manipulation.rb

```diff
@@ -52,16 +59,16 @@ def string_reverse(my_string)
 
 #######################################
 def reverse_words(my_words)
-  start = 0
-  last_index = my_words.length - 1
+  start = 0 # O(1)
+  last_index = my_words.length - 1 # O(1)
 
-  while start < last_index
-    i = start
-    until my_words[start] == " " || start > last_index
-      start += 1
+  while start < last_index # loop runs n times
+    i = start # O(1)
+    until my_words[start] == " " || start > last_index  # happens n times, not sure
```

Comment:
To figure out the time complexity, consider a character at any index and track how many times it gets visited. 
- All characters that are spaces will only get visited once - in the loop starting on line 67.
- Any character that's part of a word, will get visited once when the beginning and end of the word is getting determined.
- After the begin and end of the word is determined, each character in the word will get visited again while the word is getting reversed. 
So, in total, each character in each word is getting visited twice: once while the word boundary is getting determined, followed once while the word is getting reversed. This will bring the time complexity to O(n) + O(n).

Simply looking at loop conditionals is insufficient because although there are n characters between start and last_index in the beginning, the inner loops adds complexity. start gets updated from 0 to end of the first word in the first loop. So, for one iteration through the outer loop, start would have jumped from beginning of first character in my_words to the character following the first word.